ci/docker_run.sh

```diff
@@ -17,11 +17,26 @@ fi
 
 echo "Running Docker CI build for '$IMAGE_NAME' "
 
-docker build  -t $IMAGE_NAME .
+# Remove old docker cid just in case
+rm -f docker_cid
+
+docker build -t $IMAGE_NAME .
 exit_code=$?; [[ $exit_code != 0 ]] && exit $exit_code
+
+cleanup() {
+  echo "Running cleanup..."
+  echo "Killing running containers..."
+  cat docker_cid | xargs docker kill 
+  echo "Cleaning stopped docker containers and dangling images/networks/volumes..."
+  docker system prune -f || true
+  echo "Cleanup complete."
+}
+trap cleanup EXIT
+
 # Run the command, skip the first argument, which is the image name
-docker run --sig-proxy=true --rm $IMAGE_NAME ${@:2}
+echo "Running tests in built docker image"
+docker run --sig-proxy=true --cidfile=docker_cid --rm $IMAGE_NAME ${@:2}
```

Comment:
won't this file get overwritten due to our running multiple containers per build ?

Reply:
Looks like we get a separate workspace per run.

Reply:
nvmd... each job / container launch has its own dir.